lib/publiccloud/instance.pm

```diff
@@ -477,4 +481,166 @@ sub network_speed_test() {
+
+    my $url = get_var('PUBLIC_CLOUD_PERF_DB_URI');
+    return unless ($url);
+
+    my $token = get_var('_PUBLIC_CLOUD_PERF_DB_TOKEN');
+    return unless ($token);
+
+    my $org = get_var('PUBLIC_CLOUD_PERF_DB_ORG', 'qec');
+    my $db = get_var('PUBLIC_CLOUD_PERF_DB', 'perf');
+
+    my $tags = {
+        instance_type => get_required_var('PUBLIC_CLOUD_INSTANCE_TYPE'),
+        os_build => get_required_var('BUILD'),
+        os_flavor => get_required_var('FLAVOR'),
+        os_version => get_required_var('VERSION'),
+        os_kernel_release => $results->{kernel_release},
+        os_kernel_version => $results->{kernel_version},
+    };
+
+    $tags->{os_pc_build} = get_var('PUBLIC_CLOUD_QAM') ? 'N/A' : get_var('PUBLIC_CLOUD_BUILD', 0);
+    $tags->{os_pc_kiwi_build} = get_var('PUBLIC_CLOUD_QAM') ? 'N/A' : get_var('PUBLIC_CLOUD_BUILD_KIWI', 0);
+
+    record_info("STORE analyze", 'bootup');
+    # Store values in influx-db
+
+    my $data = {
+        table => 'bootup',
+        tags => $tags,
+        values => $results->{analyze}
+    };
+    influxdb_push_data($url, $db, $org, $token, $data);
+
+    record_info("STORE blame", $results->{type});
+    $tags->{boottype} = $results->{type};
+    $data = {
+        table => 'bootup_blame',
+        tags => $tags,
+        values => $results->{blame}
+    };
+    influxdb_push_data($url, $db, $org, $token, $data);
+}
+
+sub systemd_time_to_second
+{
+    my $str_time = trim(shift);
+    if ($str_time !~ /^(?<check_min>(?<min>\d{1,2})\s*min\s*)?((?<sec>\d{1,2}\.\d{1,3})s|(?<ms>\d+)ms)$/) {
+        die("Unable to parse systemd time '$str_time'");
+    }
+    my $sec = $+{sec} // $+{ms} / 1000;
+    $sec += $+{min} * 60 if (defined($+{check_min}));
+    return $sec;
+}
+
+sub extract_analyze_time {
+    my $str_time = shift;
+    my $res = {};
+    ($str_time) = split(/\r?\n/, $str_time, 2);
+    $str_time =~ s/Startup finished in\s*//;
+    $str_time =~ s/=(.+)$/+$1 (overall)/;
+    for my $time (split(/\s*\+\s*/, $str_time)) {
+        $time = trim($time);
+        my ($time, $type) = $time =~ /^(.+)\s*\((\w+)\)$/;
+        $res->{$type} = systemd_time_to_second($time);
+    }
+    map { die("Fail to detect $_ timing") unless exists($res->{$_}) } qw(kernel initrd userspace overall);
+    return $res;
+}
+
+sub extract_blame_time {
+    my $str_time = shift;
+    my $ret = {};
+    for my $line (split(/\r?\n/, $str_time)) {
+        $line = trim($line);
+        my ($time, $service) = $line =~ /^(.+)\s+(\S+)$/;
+        $ret->{$service} = systemd_time_to_second($time);
+    }
+    return $ret;
+}
+
+sub do_systemd_analyze_time {
+    my ($instance, %args) = @_;
+    $args{timeout} = 120;
+    my $start_time = time();
+    my $output = "";
+    my @ret;
+
+    # Wait for guest register, before calling syastemd-analyze
+    $instance->wait_for_guestregister();
+    while ($output !~ /Startup finished in/ && time() - $start_time < $args{timeout}) {
```

Comment:
no loop needed because we not waiting for anything here , we just measuring system state **when measurement was called** and other code will make sure that it is done in proper time and place

Reply:
After **loop** for Startup finished removed, I got many test rerun Failing for:
```
Bootup is not yet finished (org.freedesktop.systemd1.Manager.FinishTimestampMonotonic=0).
Please try again later.
Hint: Use 'systemctl list-jobs' to see active jobs
SCRIPT_FINISHEDRo5gS-1-
```
see: 
[3723](http://10.168.4.117/tests/3723) 
[3719](http://10.168.4.117/tests/3719) 
[3721](http://10.168.4.117/tests/3721)

It seems that the measure needs for some  wait-sync before to start, so I restore that loop and retry

Reply:
see note https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/16577#issuecomment-1460886997